crypto/provider_conf.c

```diff
@@ -130,6 +191,13 @@ static int prov_already_activated(const char *name,
     return 0;
 }
 
+/*
+ * Attempt to activate a provider
+ * Returns:
+ * 1 on successful activation
+ * 0 on failed activation for non-fatal error
+ * < 0 on failed activation for fatal errors
```

Comment:
It looks like there are several `return 0` statements at the beginning of this function which are probably really fatal and should be -1.

Reply:
fixed.  Note the soft provider setting isn't documented as far as I can see, but seems to imply that failure to allocate a provider should be treated as non-fatal